hyperspy/tests/signal/test_assign_subclass.py

```diff
@@ -1,69 +1,236 @@
 from nose.tools import assert_true
 
+import numpy as np
+
 import hyperspy.api as hs
 from hyperspy.io import assign_signal_subclass
 
 
-def test_signal():
-    assert_true(assign_signal_subclass(
-        record_by="",
-        signal_type="",
-        signal_origin="") is hs.signals.BaseSignal)
+class TestSignalAssignSubclass:
+
+    def test_signal(self):
+        assert_true(assign_signal_subclass(
```

Comment:
You could use `assert_is` instead. The same goes for all other tests in this class.